test/network_autoyast_test.rb

```diff
@@ -148,4 +148,66 @@
       network_autoyast.merge_configs("dns" => {}, "routing" => {}, "devices" => {})
     end
   end
+
+  describe "#set_network_service" do
+    Yast.import "NetworkService"
+    Yast.import "Mode"
+
+    let(:network_autoyast) { Yast::NetworkAutoYast.instance }
+
+    before(:each) do
+      allow(Yast::Mode).to receive(:autoinst).and_return(true)
+    end
+
+    def product_use_nm(nm_used)
```

Comment:
I think the spec is more readable if helpers like this are named with the stub/mock prefix, like `allow_product_use_nm`, `allow_networking_section`.